client/src/components/ElementEditor/HoverBar.scss

```diff
@@ -1,65 +1,80 @@
-.element-editor__add-block-area-container {
-  height: 0;
-  display: flex;
-  width: 100%;
-  position: relative;
-  align-items: center;
-}
+$transition: all ease .2s;
 
-.element-editor__add-block-area {
-  background-color: transparent;
-  min-height: $spacer * 1.5;
-  width: 100%;
-  margin: 0;
-  padding: 0;
-  border: 0;
-  outline: 0;
-  transition: 0s;
-}
+.element-editor {
+  &__hover-bar {
+    height: 0;
+    display: flex;
+    width: 100%;
+    position: relative;
+    align-items: center;
 
-.element-editor__add-block-hover-bar {
-  align-self: center;
-  width: 100%;
-  -webkit-appearance: button;
-  border: 0;
-  position: absolute;
-  display: flex;
-  justify-content: flex-end;
-  padding: 0;
-  top: 50%;
-  transform: translateY(-66%);
+    &__area {
```

Comment:
I think this level of nesting violates BEM a little... you should be able to list these separately, e.g.

```scss
.element-editor {
  &__hover-bar { ... }
  &__area { ... }
}
```

We try and keep nesting levels to a minimum wherever possible (extreme example is the code you've deleted in this file). The next block could also be removed, e.g. `&__area { &-inner { ... } }` becomes `&__area { ... } &__area-inner { ... }`.

Reply:
I'm happy to fix it as suggested, but isn't the underlying problem here that all those components are prefixed with element-editor?

It's probably too late to reconsider this, but I think it would make more sense to say that the root of the component should be the top block and parts of the component becomes elements and get the `__` seperator.

Reply:
Yeah I think that's a conversation for another issue - we can't change these class names any more without breaking things for people (potentially)